sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/connectionstring/ConnectionSettingsFilter.java

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation.connectionstring;
+
+
+/**
+ * Type represents contract for applying filtering on {@link ConnectionSettings}.
+ */
+@FunctionalInterface
+interface ConnectionSettingsFilter {
+    /**
+     * Applies the filter function to the given connection settings.
+     *
+     * @param inputSettings the input settings
+     * @return the result of filtering
+     */
+    ConnectionSettings apply(ConnectionSettings inputSettings);
+
+    static ConnectionSettingsFilter allRequired(final String... settingNames) {
+        return (ConnectionSettings inputSettings) -> {
+            ConnectionSettings outputSettings = inputSettings.clone();
+            for (final String settingName : settingNames) {
+                if (outputSettings.hasSetting(settingName)) {
+                    outputSettings.removeSetting(settingName);
+                } else {
+                    return null;
+                }
+            }
+            return outputSettings;
+        };
+    }
+
+    static ConnectionSettingsFilter optional(final String... settingNames) {
+        return (ConnectionSettings inputSettings) -> {
+            ConnectionSettings outputSettings = inputSettings.clone();
+            for (final String settingName : settingNames) {
+                if (outputSettings.hasSetting(settingName)) {
```

Comment:
Could we just call remove here? Do we need to check for it existing in the map before removing?

Reply:
good catch, this can be removed.